docs: added doc support for file-addition-trigger (#1240)
harness/canary: apps/portal/src/content/docs/components/file-additions-trigger.mdx

---

> title: File Additions Trigger

Reviewer: ```suggestion
title: FileAdditionsTrigger
```

> description: File Additions Trigger component documentation

Reviewer: ```suggestion
description: File Additions Trigger component
```

---

> The File Additions Trigger component provides a UI element for creating new files or uploading files. It uses a dropdown menu to offer these options.

Reviewer: ```suggestion
The `FileAdditionsTrigger` component provides a UI element for creating new files or uploading files. It uses a dropdown menu to offer these options.
```

import { DocsPage } from "@/components/docs-page";

<DocsPage.ComponentExample
  client:only
  code={`<FileAdditionsTrigger

>   useTranslationStore={() => ({ t: (key, fallback) => fallback })}
>   pathNewFile="/new-file"
>   pathUploadFiles="/upload-files"

Reviewer: The formatting of the `ComponentExample` code highlighting is a bit off an expects 4 spaces instead of 2. 2 is fine everywhere else
```suggestion
    useTranslationStore={() => ({ t: (key, fallback) => fallback })}
    pathNewFile="/new-file"
    pathUploadFiles="/upload-files"
```

/>`}
/>

## Usage

```typescript jsx
import { FileAdditionsTrigger } from '@harnessio/ui/components'

> 

Reviewer: Remember to add the ellipsis to indicate that this isn't a fully working example.
```suggestion

// ...

```

return (
  <FileAdditionsTrigger
    useTranslationStore={useTranslationStoreHook}  // [REQUIRED] Translation hook
    pathNewFile="/new-file"                         // [REQUIRED] New file path

>     pathUploadFiles="/upload-files"                // [REQUIRED] Upload files path

Reviewer: We can drop `[REQUIRED]` for non-optional props (also only include these style comments in the _"API Reference"_ section)

>     useTranslationStore={useTranslationStoreHook}  // [REQUIRED] Translation hook
>     pathNewFile="/new-file"                         // [REQUIRED] New file path
>     pathUploadFiles="/upload-files"                // [REQUIRED] Upload files path
>     className="my-class"                           // [OPTIONAL] Additional CSS classes

Reviewer: You can leave the comments for the _"API Reference"_ section.
```suggestion
    useTranslationStore={useTranslationStoreHook}
    pathNewFile="/new-file"
    pathUploadFiles="/upload-files"
    className="my-class"
```

>     className="my-class"                           // [OPTIONAL] Additional CSS classes

Reviewer: This component doesn't accept a `className` prop
```suggestion
```

  />
)
```

> ## API Reference

Reviewer: For single-part components, I've been adding a typical usage example in the _"Usage"_ section and an all props example in the _"API Reference"_ section along with a short paragraph explaining things a bit more. This component is pretty simple though and there are no optional props, so it would be the same example.

Author: Sure, will follow the same for other docs

<DocsPage.PropsTable
  props={[
    {
      name: "useTranslationStore",
      description:
        "Function that returns a translation store object with a 't' function for internationalization support. The 't' function takes a key and fallback text.",
      required: true,
      value: "() => { t: (key: string, fallback: string) => string }",
    },
    {
      name: "pathNewFile",
      description:
        "URL path to navigate to when the user selects the 'Create New File' option from the dropdown",
      required: true,
      value: "string",
    },
    {
      name: "pathUploadFiles",
      description:
        "URL path to navigate to when the user selects the 'Upload Files' option from the dropdown",
      required: true,
      value: "string",
    },

>     {
>       name: "className",
>       description: "Additional CSS classes to apply to the component",
>       required: false,
>       value: "string",
>     },

Reviewer: This component doesn't accept a `className` prop.
```suggestion
```

---
title: FileAdditionsTrigger
description: FileAdditionsTrigger component
---

The `FileAdditionsTrigger` component provides a UI element for creating new files or uploading files. It uses a dropdown menu to offer these options.

import { DocsPage } from "@/components/docs-page";

<DocsPage.ComponentExample
  client:only
  code={`<FileAdditionsTrigger
    useTranslationStore={() => ({ t: (key, fallback) => fallback })}
    pathNewFile="/new-file"
    pathUploadFiles="/upload-files"
/>`}
/>

## Usage

```typescript jsx
import { FileAdditionsTrigger } from '@harnessio/ui/components'

// ...

return (
  <FileAdditionsTrigger
    useTranslationStore={useTranslationStoreHook}
    pathNewFile="/new-file"
    pathUploadFiles="/upload-files"
  />
)
```

## API Reference

The `FileAdditionsTrigger` component helps in navigation to create new files or upload files.

```typescript jsx
<FileAdditionsTrigger

>   useTranslationStore={useTranslationStoreHook}
>   pathNewFile="/new-file"
>   pathUploadFiles="/upload-files"

Reviewer: Can we add a quick comment for each of the props to briefly explain them?
```suggestion
  useTranslationStore={useTranslationStoreHook} // something interesting
  pathNewFile="/new-file"                       // something else
  pathUploadFiles="/upload-files"               // and some more
```
